multispecies_whale_detection/front_end.py

```diff
@@ -0,0 +1,167 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Ops for signal processing that prepares neural network input."""
+import dataclasses
+
+import tensorflow as tf
+import tensorflow_probability as tfp
+
+
+def amplitude_ratio_to_db(x):
+  return 20.0 * tf.math.log(x / 10.0)
+
+
+def db_to_amplitude_ratio(x):
+  return tf.math.pow(10.0, x / 20.0)
+
+
+@dataclasses.dataclass(frozen=True)
+class FrontEndConfig:
+  """Configuration data object for the FrontEnd Keras layer."""
+
+  sample_rate: int = 2000
+  """Sample rate of the input waveform."""
```

Comment:
piconit: Why not a single-line comment? (For some reason putting the comment above the variable also seems more intuitive to me, but no hard preference.)